crates/oxc_transformer/src/typescript/namespace.rs

```diff
@@ -11,99 +11,105 @@ use oxc_syntax::operator::{AssignmentOperator, LogicalOperator};
 // 1. register scope for the newly created function: <https://github.com/babel/babel/blob/08b0472069cd207f043dd40a4d157addfdd36011/packages/babel-plugin-transform-typescript/src/namespace.ts#L38>
 impl<'a> TypeScript<'a> {
     // `namespace Foo { }` -> `let Foo; (function (_Foo) { })(Foo || (Foo = {}));`
-    pub(super) fn transform_statements_for_namespace(&self, stmts: &mut Vec<'a, Statement<'a>>) {
+    pub(super) fn transform_program_for_namespace(&self, program: &mut Program<'a>) {
+        if !has_namespace(program.body.as_slice()) {
```

Comment:
I was confused about this, you should comment on namespace can only be at top level.